internal/webhook/v1/alertmanager_config_secret_webhook.go

```diff
@@ -54,25 +52,30 @@ func SetupAlertmanagerConfigSecretWebhookWithManager(mgr ctrl.Manager) (err erro
 // Modifying the path for an invalid path can cause API server errors; failing to locate the webhook.
 // +kubebuilder:webhook:path=/validate-alertmanager-config,mutating=false,failurePolicy=fail,sideEffects=None,groups="",resources=secrets,verbs=create;update,versions=v1,name=vsecret-v1.kb.io,admissionReviewVersions=v1
 
-// SecretCustomValidator struct is responsible for validating the Secret resource
+// AlertmanagerConfigSecretValidator struct is responsible for validating the Secret resource
 // when it is created, updated, or deleted.
 //
 // NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
 // as this struct is used only for temporary operations and does not need to be deeply copied.
-
-type AlertmanagerConfigSecretCustomValidator struct {
+// +kubebuilder:object:generate=false
+type AlertmanagerConfigSecretValidator struct {
 	client client.Client
 }
 
-var _ webhook.CustomValidator = &AlertmanagerConfigSecretCustomValidator{}
+var _ webhook.CustomValidator = &AlertmanagerConfigSecretValidator{}
 
 // ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type Secret.
-func (v *AlertmanagerConfigSecretCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+func (v *AlertmanagerConfigSecretValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
 	secret, ok := obj.(*corev1.Secret)
 	if !ok {
 		return nil, fmt.Errorf("expected a Secret object but got %T", obj)
 	}
 
+	// Only validate secrets that are specifically marked as alertmanager-config
+	if !v.isAlertmanagerConfigSecret(secret) {
+		return nil, nil
+	}
+
```

Comment:
How did this work before ? How could this webhook not have disturbed secret creation until  now ?

Reply:
Thanks to the validation webhook configuration: https://github.com/giantswarm/observability-operator/blob/f2d13f901f0bab05f283564a793a20ab547102a7/helm/observability-operator/templates/webhook/validatingwebhookconfiguration.yaml#L23